ui/timeline/src/commonMain/kotlin/social/androiddev/timeline/ImmutableListWrapper.kt

```diff
@@ -0,0 +1,29 @@
+/*
+ * This file is part of Dodo.
+ *
+ * Dodo is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version.
+ *
+ * Dodo is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty
+ * of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with Dodo.
+ * If not, see <https://www.gnu.org/licenses/>.
+ */
+package social.androiddev.timeline
+
+import androidx.compose.runtime.Immutable
+
+/**
+ *  The Compose Compiler cannot infer the stability of a parameter if a List is used in it,
+ *  even if the item type is stable. This wrapper class uses the [Immutable] annotation
+ *  to promise the compiler it is indeed Immutable.
+ */
+@Immutable
+data class ImmutableListWrapper<T : Any?>(
+    val items: List<T>
+) {
```

Comment:
FYI for those of you will be writing more compose UI and passing `List`

Reply:
alternatively to a data class wrapper, we can use kotlinx immutable collections https://github.com/Kotlin/kotlinx.collections.immutable 

Some info here https://medium.com/androiddevelopers/jetpack-compose-stability-explained-79c10db270c8

Reply:
Yeah - I tried importing it and got failed builds so quickly did my own wrapper for now. Let me get this PR merged in and then can circle back to that so I don't have to keep dealing with huge conflicts